### PR TITLE
Let civilian buildings spawn civilians on death

### LINKED ISSUE #12240: Make ^CivBuilding emit civillians when destroyed.
Following @Micr0Bit's comment in #12174.

> and we could make this a bit more interesting by having ^CivBuilding emit "civs" ... when they are destroyed ... with "EjectOnDeath:"
> 
> so we can actually still see "Civilians" ingame

### PR DESCRIPTION
**Pull request by abc013**
Closes #12240.

### CODE REVIEW

File: mods/ra/rules/campaign-rules.yaml [hunk 1]
```diff
@@ -35,3 +35,8 @@ E7.noautotarget:
 
 ^Vehicle:
 	Demolishable:
+
+^CivBuilding:
+	-SpawnActorOnDeath@1:
+	-SpawnActorOnDeath@2:
+	-SpawnActorOnDeath@3:
```
Comment (abcdefg30) [inline mods/ra/rules/campaign-rules.yaml]: I'm not sure if this is necessary. However, if we want to keep it that way, you should also remove `Explodes`.

File: mods/ra/weapons/explosions.yaml [hunk 2]
```diff
@@ -212,6 +212,14 @@ UnitExplodeSmall:
 		SmudgeType: Crater
 		InvalidTargets: Structure, Wall, Trees
 
+CivBuildingExplosion:
```
Comment (reaperrr) [inline mods/ra/weapons/explosions.yaml]: Can you explain what the purpose of this and the Explodes trait on the default is?

Comment (abcdefg30) [inline mods/ra/weapons/explosions.yaml]: Otherwise the civilians won't panic, but just stand still (on top of each other, if multiple spawned).

Comment (abcdefg30) [inline mods/ra/weapons/explosions.yaml]: Otherwise the civilians won't panic, but just stand still (on top of each other, if multiple spawned).

Comment (reaperrr) [inline mods/ra/weapons/explosions.yaml]: Ok.

Comment (pchote) [inline mods/ra/weapons/explosions.yaml]: Can you please add a comment explaining this?

**Comment by abc013** [comment]
Updated.

File: mods/ra/weapons/explosions.yaml [hunk 3]
```diff
@@ -212,6 +212,15 @@ UnitExplodeSmall:
 		SmudgeType: Crater
 		InvalidTargets: Structure, Wall, Trees
 
+# Used to panic civilians which are emitted from a killed CivBuilding
+CivBuildingExplosion:
+	Projectile: Bullet
+		Speed: 1c682
```
Comment (reaperrr) [inline mods/ra/weapons/explosions.yaml]: You can remove the projectile, `Explodes` calls the impact directly.

**Comment by abc013** [comment]
Updated.

**Comment by Micr0Bit** [comment]
:+1:

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/824e529f088de10690d1282b0865308ed3f11d2a...5cd63f464dba99316b0d5094dd8e21f009f19e0d)

**Pull request merged**